src/logging.cpp

```diff
@@ -276,14 +320,43 @@ void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& loggi
         // reopen the log file, if requested
         if (m_reopen_file) {
             m_reopen_file = false;
-            FILE* new_fileout = fsbridge::fopen(m_file_path, "a");
-            if (new_fileout) {
-                setbuf(new_fileout, nullptr); // unbuffered
-                fclose(m_fileout);
-                m_fileout = new_fileout;
-            }
+            ReopenFile();
         }
         FileWriteStr(str_prefixed, m_fileout);
+        m_file_size += str_prefixed.size();
+
+        // If debug.log is large, rotate it (rename and recreate).
+        if (m_rotate_keep > 0 && m_file_size > m_file_limit * 1e6) {
```

Comment:
What if I want to rotate, but never delete?

Reply:
> What if I want to rotate, but never delete?

`-debuglogrotatekeep=9999999` or is that ugly? Should there be a separate `-debuglogenable` boolean (default true; if set to false, behave as today), and then `-debuglogrotatekeep=0` could mean "never delete". Would that be preferable? Or something else? Thanks for the comments!